Server/Ebenezer/User.cpp

```diff
@@ -8889,6 +8880,110 @@ bool CUser::GetStartPosition(int16_t* x, int16_t* z)
 	return false;
 }
 
+void CUser::SendClassPromotion(int16_t newClass)
+{
+	constexpr uint8_t classPromotionOpcode = 0x06;
+	char send_buff[32] = {};
+	int send_index = 0;
+
+	m_pUserData->m_sClass = newClass;
+	SetUserAbility();
+	Send2AI_UserUpdateInfo();
+	UserDataSaveToAgent();
+
+	SetByte(send_buff, WIZ_CLASS_CHANGE, send_index);
+	SetByte(send_buff, classPromotionOpcode, send_index);
+	SetShort(send_buff, m_pUserData->m_sClass, send_index);
+	SetShort(send_buff, _socketId, send_index);
+	Send(send_buff, send_index);
+	if (m_pMain != nullptr)
+		m_pMain->Send_Region(send_buff, send_index, m_pUserData->m_bZone, m_RegionX, m_RegionZ, this);
+
+	if (m_sPartyIndex != -1)
```

Comment:
This isn't specifically a problem with this line, but what we're missing -- as compared to official -- is the clan update.

It needs to be double-checked as to whether it's truly necessary, because the clan update (`CKnightsManager::CurrentKnightsMember()`) specifically only updates our clan member list -- it doesn't update our class for other clan members, as you might expect to be the case.

You would think that `WIZ_CLASS_CHANGE`/`NOVICE_CLASS_CHANGE_REQ` should update (our) clan member list already, so that would need to be checked to see if it does that officially. The update may be redundant if so.